src/drivers/driver.ts

```diff
@@ -69,6 +70,12 @@ export abstract class CMakeDriver implements vscode.Disposable {
    * @returns The exit code from CMake
    */
   protected abstract doConfigure(extra_args: string[], consumer?: proc.OutputConsumer): Promise<number>;
+  protected abstract doCacheConfigure(): Promise<number>;
+
+  protected _isConfiguredAtLeastOnce = false;
```

Comment:
You only set this to true in the file-api driver, but it is owned by the base driver class.  I think you should make this private (the 'get' below can be protected), and set it to true when configure is called here in the base class.